frontend/javascripts/admin/dataset/dataset_add_zarr_view.tsx

```diff
@@ -0,0 +1,217 @@
+import { Form, Input, Button, Col, Row, Divider } from "antd";
+import { connect } from "react-redux";
+import React, { useState } from "react";
+import type { APIUser } from "types/api_flow_types";
+import type { OxalisState } from "oxalis/store";
+import { exploreRemoteDataset, storeRemoteDataset } from "admin/admin_rest_api";
+import messages from "messages";
+import { jsonStringify } from "libs/utils";
+import { CardContainer } from "admin/dataset/dataset_components";
+import Password from "antd/lib/input/Password";
+import TextArea from "antd/lib/input/TextArea";
+import { AsyncButton } from "components/async_clickables";
+import Toast from "libs/toast";
+import DataLayer from "oxalis/model/data_layer";
+import _ from "lodash";
+import { Hint } from "oxalis/view/action-bar/download_modal_view";
+const FormItem = Form.Item;
+
+type OwnProps = {
+  onAdded: (arg0: string, arg1: string) => Promise<void>;
+};
+type StateProps = {
+  activeUser: APIUser | null | undefined;
+};
+type Props = OwnProps & StateProps;
+
+function DatasetAddZarrView(props: Props) {
+  const { activeUser, onAdded } = props;
+  const [datasourceConfig, setDatasourceConfig] = useState<string>();
+  // const [exploreLog, setExploreLog] = useState<string>();
+  const [datasourceUrl, setDatasourceUrl] = useState<string>("");
+  const [usernameOrAccessKey, setUsernameOrAccessKey] = useState<string>("");
+  const [passwordOrSecretKey, setPasswordOrSecretKey] = useState<string>("");
+  const [selectedProtocol, setSelectedProtocol] = useState<string>("https://");
+
+  function validateUrls(userInput: string) {
+    if (
+      (userInput.indexOf("https://") === 0 && userInput.indexOf("s3://") !== 0) ||
+      (userInput.indexOf("https://") !== 0 && userInput.indexOf("s3://") === 0)
+    ) {
+      throw new Error("Dataset URL must employ either the https:// or s3:// protocol.");
+    } else {
+      setSelectedProtocol(userInput.indexOf("https://") === 0 ? "https://" : "s3://");
+    }
+  }
+
+  async function handleExplore() {
+    if (datasourceUrl) {
+      let datasourceToMerge;
+      if (!usernameOrAccessKey || !passwordOrSecretKey) {
+        datasourceToMerge = await exploreRemoteDataset([datasourceUrl]);
+      } else {
+        datasourceToMerge = await exploreRemoteDataset([datasourceUrl], {
+          username: usernameOrAccessKey,
+          pass: passwordOrSecretKey,
+        });
+      }
+      if (datasourceToMerge) {
+        if (datasourceConfig) {
+          // TODO: check that both datasources have same voxel size else warning
+          let currentDatasource;
+          try {
+            currentDatasource = JSON.parse(datasourceConfig);
+          } catch (e) {
+            Toast.error("The loaded datasource config contains invalid JSON.");
+            return;
+          }
+          const layers = currentDatasource.dataLayers.concat(datasourceToMerge.dataLayers);
+          const uniqueLayers = _.uniqBy(layers, (layer: DataLayer) => layer.name);
```

Comment:
The name attribute comes from the layer uris. If they are explored from the same source, they are already unique. 
If the backend receives multiple uris, it employs a logic to uniquiefy them (by adding `_2`, `_3`, etc),

If the backend route is called multiple times, the frontend should probably employ the same logic

Reply:
> If the backend route is called multiple times, the frontend should probably employ the same logic when merging

oh, sounds good. @Dagobert42 let me know in case you need some guidance to implement this :)